### PR TITLE
LibJS: Remove the last 2 old non-standard Object helper methods

### PR DESCRIPTION
**Pull request by IdanHo**
Also fixes 17 test262 test cases as an extra bonus :^)

(The first 2 commits are issues in the implementations that were uncovered after using the correct object helpers)

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Object.cpp [hunk 1]
```diff
@@ -190,6 +190,18 @@ bool Object::create_data_property_or_throw(PropertyName const& property_name, Va
     return success;
 }
 
+// 7.3.6 CreateNonEnumerableDataPropertyOrThrow ( O, P, V ), https://tc39.es/proposal-error-cause/#sec-createnonenumerabledatapropertyorthrow
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Object.cpp:193]: flaw in the spec: 7.3.6 already is something else :^)
(no worries, keep it as is)

File: Userland/Libraries/LibJS/Runtime/Object.cpp [hunk 2]
```diff
@@ -190,6 +190,18 @@ bool Object::create_data_property_or_throw(PropertyName const& property_name, Va
...
+bool Object::create_non_enumerable_data_property_or_throw(PropertyName const& property_name, Value value)
+{
+    VERIFY(!value.is_empty());
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Object.cpp:196]: Let's also `VERIFY` that the `property_name` is valid.

File: Userland/Libraries/LibJS/Runtime/Object.cpp [hunk 3]
```diff
@@ -190,6 +190,18 @@ bool Object::create_data_property_or_throw(PropertyName const& property_name, Va
...
+
+    // 1. Let newDesc be the PropertyDescriptor { [[Value]]: V, [[Writable]]: true, [[Enumerable]]: false, [[Configurable]]: true }.
+    PropertyDescriptor new_description = { .value = value, .writable = true, .enumerable = false, .configurable = true };
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Object.cpp:199]: Don't know why, but lately I prefer this style:

```suggestion
    auto new_description = PropertyDescriptor { .value = value, .writable = true, .enumerable = false, .configurable = true };
```

I guess it just looks so similar :D
Just a note, no action required (unless you happen to agree)

File: Userland/Libraries/LibJS/Runtime/Array.cpp [hunk 4]
```diff
@@ -81,14 +81,6 @@ JS_DEFINE_NATIVE_GETTER(Array::length_getter)
...
         return {};
 
-    // TODO: could be incorrect if receiver/this_value is fixed or changed
-    if (!this_object->is_array()) {
-        auto value = this_object->internal_get(vm.names.length.to_string_or_symbol(), this_object);
-        if (vm.exception())
-            return {};
-        return value;
-    }
-
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Array.cpp:91]: This seems unrelated, perhaps should be its own commit with an explanation?

File: Userland/Libraries/LibJS/Runtime/Error.h [hunk 5]
```diff
@@ -20,9 +20,14 @@ class Error : public Object {
...
 
     explicit Error(Object& prototype);
+    Error(String message, Object& prototype);
+    virtual void initialize(GlobalObject&) override;
     virtual ~Error() override = default;
 
     void install_error_cause(Value options);
+
+private:
+    String m_message;
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Error.h:30]: Seems a bit wasteful to keep this around if we only use it to define a property... the previous way wasn't great but it's a bit better in that regard IMO.

**Review by linusg — COMMENTED** [review]
Awesome! I can imagine this provides a measurable interpreter start speedup (and by extension test-js speedup) already... :^)

File: Userland/Libraries/LibJS/Runtime/Array.cpp (hunk 4 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Array.cpp:91]: It is actually related, but ill split it into its own commit anyways (This is a matching removal of the same block in the setter, which used `define_property` but instead of replacing the define_property i removed both blocks as they are no longer relevant after the object rewrite.

File: Userland/Libraries/LibJS/Runtime/Error.h (hunk 5 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Error.h:30]: hmm, i guess so, will revert that.

File: Userland/Libraries/LibJS/Runtime/Object.cpp (hunk 3 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Object.cpp:199]: done :)

File: Userland/Libraries/LibJS/Runtime/Object.cpp (hunk 2 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Object.cpp:196]: added

File: Userland/Libraries/LibJS/Runtime/Array.cpp (hunk 4 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Array.cpp:91]: fixed

File: Userland/Libraries/LibJS/Runtime/Error.h (hunk 5 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/Error.h:30]: reverted

**Comment by IdanHo** [comment]
This should be ready to merge AFAICT, the ci is failing due to the memory leaking parser changes (aka unrelated)

**Comment by IdanHo** [comment]
Pushed with no changes to re trigger CI with the fixed memory leak issues
